packages/opentelemetry-sdk-trace-base/src/export/SimpleSpanProcessor.ts

```diff
@@ -64,10 +64,15 @@ export class SimpleSpanProcessor implements SpanProcessor {
       return;
     }
 
+    let exportPromise: Promise<void> | undefined = undefined;
+
     const doExport = () =>
       internal
         ._export(this._exporter, [span])
         .then((result: ExportResult) => {
+          if (exportPromise) {
+            this._pendingExports.delete(exportPromise);
```

Comment:
```suggestion
            // Ensure a second call to forceFlush won't re-export this
            this._pendingExports.delete(exportPromise);
```

Reply:
I think this wasn't clear because of reusing the existing structure to minimize the diff. I went ahead and reworked the code more to try to make the flow easier to understand. This uses an `async` function - to clarify the point of the PR, `export` itself was already async, so adding another async function should not cause async scheduling that didn't happen before.